glib/src/object.rs

```diff
@@ -3376,6 +3401,65 @@ wrapper! {
     }
 }
 
+type WeakRefNotifyData = ManuallyDrop<Pin<Box<Option<Box<dyn FnOnce() + 'static>>>>>;
+
+// rustdoc-stripper-ignore-next
+/// A handle to disconnect a weak ref notify closure.
+pub struct WeakRefNotify<T: ObjectType> {
+    object: WeakRef<T>,
+    data: WeakRefNotifyData,
+}
+
+unsafe extern "C" fn notify_func(data: ffi::gpointer, _obj: *mut gobject_ffi::GObject) {
+    let callback: Box<Option<Box<dyn FnOnce()>>> = Box::from_raw(data as *mut _);
```

Comment:
This is not very clean: you're passing a `ManuallyDrop<Pin<T>>` in here but then handle it like a `T`. It would be better to use the correct type here.

Also this needs a comment explaining why it is safe to drop the closure from the callback here (i.e. why it is guaranteed to be called only once).